src/proof/veriT/veriT_post_processor.cpp

```diff
@@ -0,0 +1,1163 @@
+/*********************                                                        */
+/*! \file veriT_post_processor.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Hanna Lachnitt
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief The module for processing proof nodes into veriT proof nodes
+ **/
+
+#include "proof/veriT/veriT_post_processor.h" 
+#include <cstdlib>
+#include <functional>
+#include <memory>
+#include <vector>
+
+namespace CVC4 {
+
+namespace proof {
+
+VeriTProofPostprocessCallback::VeriTProofPostprocessCallback(ProofNodeManager* pnm):d_pnm(pnm){
+  d_nm = NodeManager::currentNM();
+}
+
+void VeriTProofPostprocessCallback::initializeUpdate(){}
+
+bool VeriTProofPostprocessCallback::shouldUpdate(std::shared_ptr<ProofNode> pn,
+				 		 bool& continueUpdate){
+  //In case of a 1-1 correspondence of the CVC4 proof rule and the veriT proof rule the proof node is not updated.
+  //The general template for this is:
+  //
+  // ======== Proof Rule 
+  // Children: (P1 ... Pn)
+  // Arguments: (A1 ... An)
+  // ---------------------
+  // Conclusion: (C)
+  // 
+  // proof rule: VeriT proof rule
+  // proof term: (C)
+  // premises: (P1 ... Pn)
+  // args: ()
+  //
+  // I.e. only the proof rule changes. The veriTPrinter will map any remaining CVC4 rule to the corresponding VeriT rule during printing.
+  // TEMP: For now translate all!
+	
+  switch(pn->getRule()){
+    case PfRule::VERIT_RULE: return false;
+    default: return true;
+  }
+} 
+
+//TEMP: ad-hoc code for printing the current proof node 
+static void printCurrentNode(Node res,
+			     PfRule id,
+			     const std::vector<Node>& children,
+			     const std::vector<Node>& args){
+
+  std::cout << "Print current node information" <<std::endl;
+  std::cout << "  (" << id << ",{";
+  
+  if(children.size() > 0)
+    std::cout << children[0];
+
+  for(int i=1;i<children.size();i++){
+    std::cout << "," << children[i];
+  }
+  std::cout << "},{";
+
+  if(args.size() > 0)
+    std::cout << args[0];
+
+  for(int i=1;i<args.size();i++){
+    std::cout << "," << args[i];
+  }
```

Comment:
Just FYI, you can give vector of nodes directly to the printer. They will be printed as `[ n0 ... nm ]`.